Library/Homebrew/cmd/update-report.rb

```diff
@@ -289,6 +291,34 @@ def link_completions_manpages_and_docs(repository = HOMEBREW_REPOSITORY)
         #{e}
     EOS
   end
+
+  def migrate_gcc_dependents_if_needed
+    return if OS.mac?
+
+    # TODO: Remove this block when GCC 12 ships.
+    begin
+      return if Formula["gcc"].version < 12
+    rescue FormulaUnavailableError
+      return if Homebrew::EnvConfig.install_from_api?
+    end
```

Comment:
Does the GCC formula version actually matter considering #13631 should add it regardless?

Reply:
Good question. I thought it would be useful to wait until it actually ships, but I think it might be helpful to give it a try before then.

I'll remove this.